tests/test.py

```diff
@@ -512,6 +514,457 @@ def testZstashWithHPSS(self):
+        # Put the original index.db back.
+        os.remove('zstash/index.db')
+        shutil.copy('zstash/index_old.db', 'zstash/index.db')
+        os.chdir('../')
+
+        print('Verifying the data from database with the actual files')
+        # Checksums from HPSS
+        cmd = ['sqlite3', 'zstash_test/zstash/index.db', 'SELECT md5, name FROM files;']
+        output_hpss, err_hpss = run_cmd(cmd)
+        hpss_dict = {}
+
+        for l in output_hpss.split('\n'):
+            l = l.split('|')
+            if len(l) >= 2:
+                f_name = l[1]
+                f_hash = l[0]
+                hpss_dict[f_name] = f_hash
+
+        # Checksums from local files
+        cmd = '''find zstash_test_backup -regex .*\.txt.* -exec md5sum {} + '''
+        output_local, err_local = run_cmd(cmd)
+        local_dict = {}
+
+        for l in output_local.split('\n'):
+            l = l.split('  ')
+            if len(l) >= 2:
+                f_name = l[1].split('/')  # remove the 'zstash_test_backup'
+                f_name = '/'.join(f_name[1:])
+                f_hash = l[0]
+                local_dict[f_name] = f_hash
+        print('filename|HPSS hash|local file hash')
+        for k in local_dict:
+            print('{}|{}|{}'.format(k, hpss_dict[k], local_dict[k]))
+        
+        cleanup(hpss_path)
+
+    def testKeepTars(self):
+        print('*' * 40)
+        print('testKeepTars')
+        print('*' * 40)
+        if os.path.exists('test_files'):
+            shutil.rmtree('test_files')
+        os.mkdir('test_files')
+        run_cmd('touch test_files/file1.txt')
+        run_cmd('touch test_files/file2.txt')
+        hpss_path = 'none'
+        # Run `zstash create`
+        run_cmd('zstash create --hpss={} test_files'.format(hpss_path))
+        actual = sorted(os.listdir('test_files/zstash/'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqual(actual, expected)
+        os.chdir('test_files')
+        # Delete txt files
+        run_cmd('rm file1.txt file2.txt')
+        # Run `zstash extract`
+        output, err = run_cmd('zstash extract --hpss={}'.format(hpss_path))
+        # Run `zstash check`
+        output, err = run_cmd('zstash check --hpss={}'.format(hpss_path))
+        self.assertEqual(output+err, 'INFO: Opening tar archive zstash/000000.tar\nINFO: Checking file1.txt\nINFO: Checking file2.txt\nINFO: No failures detected when checking the files.\n')
+        # Check that tar and db files were not deleted
+        actual = sorted(os.listdir('zstash/'))
+        expected = sorted(['000000.tar', 'index.db'])
+        self.assertEqual(actual, expected)
+        # Check that tar file is read-only
+        # https://stackoverflow.com/questions/1861836/checking-file-permissions-in-linux-with-python
+        stat = os.stat('zstash/000000.tar')
+        oct_mode = str(oct(stat.st_mode))[-3:]
+        self.assertEqual(oct_mode, '444')
```

Comment:
I think there are named flags in https://docs.python.org/3/library/os.html#os.chmod but I figured `444` was fine for read-only for everyone.